src/app/qgsdatasourcemanagerdialog.h

```diff
@@ -0,0 +1,74 @@
+/***************************************************************************
+    qgsdatasourcemanagerdialog.h - datasource manager dialog
+
+    ---------------------
+    begin                : May 19, 2017
+    copyright            : (C) 2017 by Alessandro Pasotti
+    email                : apasotti at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSDATASOURCEMANAGERDIALOG_H
+#define QGSDATASOURCEMANAGERDIALOG_H
+
+#include <QList>
+#include <QDialog>
+#include "qgsoptionsdialogbase.h"
+#include "qgsguiutils.h"
+
+class QgsBrowserDockWidget;
+class QgsRasterLayer;
+class QgsMapCanvas;
+
+namespace Ui
+{
+  class QgsDataSourceManagerDialog;
+}
+
+class QgsDataSourceManagerDialog : public QgsOptionsDialogBase
```

Comment:
I would suggest that we add a new slot updateProjectHome() to QgsBrowserDockWidget. Its body would read something like this: if (mModel) mModel->updateProjectHome(). Then, the connect() would be moved from QgsBrowserDockWidget::showEvent() to QgisApp constructor where the browser dock widget gets created (there would be two connects, one for each browser dock). These few changes should remove the dependency on QgisApp...